module-ballerina-openapi/annotation.bal

```diff
@@ -24,6 +24,13 @@
 # + embed - Enable auto-inject of OpenAPI documentation to current service
 # + title - Title for generated OpenAPI contract
 # + version - Version for generated OpenAPI contract
+# + description - A brief description of the API, outlining its purpose, features, and any other relevant details that help users understand what the API does and how to use it.
+# + email - The email address to contact the API provider or support.
+# + contactName - The full name of the person or organization responsible for the API.
+# + contactURL - The URL to a web page with more information about the API, the provider, or support.
+# + termOfService - The URL to the terms of service for the API.
+# + licenseName - The name of the license under which the API is provided.
+# + licenseURL - The URL to the full text of the license.
 public type ServiceInformation record {|
     string contract = "";
     string[]? tags = [];
```

Comment:
Just checking why we have marked `tags` as `string[]?`. Is this the correct usecase ? I think `string[]` would be enough with an empty array as the default value.

Reply:
This was introduced in the previous design. With this design, they might allow null values, although provide a default value. We will be able to revisit and make necessary adjustments separately.

Reply:
I think this should be revisited properly and fix. If this is nullable filed it should be represented as `string[] tags?;` rather than using `string[]?`.

Reply:
Nil means that there is no value right. That means there can be a scenario we won't have a value for this field. In that case it would be more sensible to make it optional rather than making it nillable. I think that has more ballerina-ish touch rather than marking fields nillable.

Reply:
I think this was designed sometime back and that is why we have followed this approach (maybe due to some limitation which was there at that time). Usually when designing APIs we prefer optional fields over nillable fields.

Reply:
yeah, usually we provide optional fields for recent implementation, will revisit this design and do needful.